scalahost/nsc/src/test/scala/scala/meta/tests/scalahost/mirrors/SemanticSuite.scala

```diff
@@ -229,10 +229,61 @@ class SemanticSuite extends DatabaseSuite {
   """.trim.stripMargin,
     """
       |[191..191) [Int, List[Int]]
-      |[199..199) (immutable.this.List.canBuildFrom[Int])
+      |[199..199) (scala.collection.immutable.List.canBuildFrom[Int])
       |[237..237) (C.list[Int](C.int))
       |[263..265) X.cvt[Int](*)(C.int)
       |[294..294) [C[Int]]
   """.trim.stripMargin
   )
+
+  denotations(
+    """
+      |import scala.collection.mutable.ListBuffer
+      |import scala.collection.mutable.{HashSet => HS}
+      |trait B {
+      |  type X
+      |  def x: X
+      |}
+      |class E extends B {
+      |  type X = ListBuffer[Int]
+      |  def x = ListBuffer.empty
+      |}
+      |class D extends B {
+      |  type X = HS[Int]
+      |  def x = HS.empty
+      |}
+      |object a {
+      |  val x = new E().x
+      |  val y = new D().x
+      |  def foo(b: B): b.X = {
+      |    val result = b.x
+      |    result
+      |  }
+      |}
+    """.stripMargin,
+    """
+      |<...>@340..356 => val result: b.X
+      |_empty_.B# => trait B
+      |_empty_.B#X# => abstract type X
+      |_empty_.B#x()Ljava/lang/Object;. => abstract def x: B.this.X
```

Comment:
Maybe we can use g.showCode to improve the pretty printing, but I wouldn't block a release for this.